src/components/RelicModal.tsx

```diff
@@ -403,11 +404,18 @@ function SubstatInput(props: { index: number; upgrades: RelicUpgradeValues[]; re
             placeholder="Substat"
             maxTagCount="responsive"
             options={substatOptions}
+            onChange={(e) => {
+              props.relicForm.getFieldValue(`substatType${props.index}`) ? props.relicForm.setFieldValue(`substatValue${props.index}`, 0) : props.relicForm.setFieldValue(`substatValue${props.index}`, undefined)
+              props.resetUpgradeValues()
+            }}
```

Comment:
Since we have a lot of repeats here I think moving all the `substatType${props.index}` blocks into their own variable is worth it here so we only have to write it once and then can reuse

```
  const statTypeField = `substatType${props.index}`
  const statValueField = `substatValue${props.index}`

...

            onChange={() => {
              props.relicForm.getFieldValue(statTypeField)
                ? props.relicForm.setFieldValue(statValueField, 0)
                : props.relicForm.setFieldValue(statValueField, undefined)
              props.resetUpgradeValues()
            }}
```